optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/locator/Locator.java

```diff
@@ -21,22 +21,20 @@
 import java.util.Map;
 
 import org.optaplanner.core.api.domain.locator.PlanningId;
-import org.optaplanner.core.api.domain.solution.PlanningSolution;
 import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
 import org.optaplanner.core.impl.score.director.ScoreDirector;
 
 /**
- * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  * @see PlanningId
  * @see ScoreDirector#locateWorkingObject(Object)
  */
-public class Locator<Solution_> {
+public class Locator {
 
     private final LocationStrategyResolver locationStrategyResolver;
 
     private Map<Object, Object> idToWorkingObjectMap;
 
-    public Locator(SolutionDescriptor<Solution_> solutionDescriptor) {
+    public Locator(SolutionDescriptor<?> solutionDescriptor) {
```

Comment:
The point is that this piece of code (Locator) doesn't work with Solution_ at all so the type parameter is redundant there. Also the solution descriptor is only needed to retrieve the location strategy resolver so there would be no point in casting SolutionDescriptor<?> to something else.

In fact we could replace the constructor parameter type from SolutionDescirptor to LocationStrategyResolver:

```java


    public Locator(LocationStrategyResolver locationStrategyResolver) {
        this.locationStrategyResolver = locationStrategyResolver;
    }
```

Which I think might actually be a good idea because the tests wouldn't need to construct the solution descriptor to set up the Locator.